src/network-layer/default/__tests__/RelayDefaultNetworkLayer-test.js

```diff
@@ -229,6 +237,31 @@ describe('RelayDefaultNetworkLayer', () => {
       expect(error.source).toEqual(response);
     });
 
+    it('handles server-side non-2xx  errors', () => {
+      const response = {
+        errors: [{
+          message: 'Something went completely wrong.'
+        }],
+      };
+
+      expect(fetch).not.toBeCalled();
+      networkLayer.sendMutation(request);
+      expect(fetch).toBeCalled();
+      const failureResponse = genFailureResponse(response);
+      expect(failureResponse.status >= 300 || failureResponse.status < 200).toBeTruthy();
```

Comment:
Since `failureResponse` isn't part of the network layer, let's not make assertions about it in the test body.